### PR TITLE
Implement async_graphql's input validation and guards

### LINKED ISSUE #131: [Feature]: Use async_graphql's input validation and field guards
### Tell us about your Wish

- https://async-graphql.github.io/async-graphql/en/input_value_validators.html
- https://async-graphql.github.io/async-graphql/en/field_guard.html

### Is your feature related to a bug

_No response_

### Alternatives

_No response_

### Additional Info

_No response_

### Participation

- [ ] I am willing to submit a pull request to implement this

### Disclaimer

- [X] I agree

### PR DESCRIPTION
**Pull request by lennartkloock**
## Proposed changes

- Implements async_graphql's input validation
- Implements a auth guard to reduce code duplication

Unfortunetely we can't use async_graphql's field guards for implementing the auth guard because of this: https://github.com/async-graphql/async-graphql/issues/1398

## Types of changes

What types of changes does your code introduce to Scuffle?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/ScuffleTV/scuffle/blob/main/CONTRIBUTING.md) doc
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

Closes #131

### CODE REVIEW

**Comment by TroyKomodo** [comment]
Does the schema need to be updated `mask gql prepare` ?

File: platform/api/src/api/v1/gql/guards/mod.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+// This can't be replaced by async_graphql's field guards because of this: https://github.com/async-graphql/async-graphql/issues/1398
+// I don't see a better alternative than doing this for now.
+pub async fn auth_guard<T>(
+    ctx: &Context<'_>,
+    field_name: &'static str,
+    field_value: T,
+    user_id: Ulid,
+) -> Result<T> {
```
Comment (TroyKomodo) [inline platform/api/src/api/v1/gql/guards/mod.rs:18]: Might it be worth adding the ability to specify what permissions are required?

**Comment by lennartkloock** [comment]
> Does the schema need to be updated `mask gql prepare` ?

No, no endpoints changed

File: platform/api/src/api/v1/gql/guards/mod.rs (hunk 1 above)
Comment (lennartkloock) [reply platform/api/src/api/v1/gql/guards/mod.rs:18]: Yeah, we could do that but it isn't needed right now

Comment (TroyKomodo) [reply platform/api/src/api/v1/gql/guards/mod.rs:18]: Okay, seems good then.

**Review by TroyKomodo — APPROVED** [review]
lgtm

**Pull request merged**